frontend/src/component/feature/StrategyTypes/RolloutSlider/RolloutSlider.tsx

```diff
@@ -142,20 +195,47 @@ const RolloutSlider = ({
                     }
                 />
             </StyledBox>
-            <StyledSlider
-                min={0}
-                max={100}
-                value={value}
-                getAriaValueText={valuetext}
-                aria-labelledby='discrete-slider-always'
-                step={1}
-                data-testid={ROLLOUT_SLIDER_ID}
-                marks={marks}
-                onChange={onChange}
-                valueLabelDisplay='on'
-                disabled={disabled}
-            />
-        </div>
+            <StyledSliderContainer>
+                <SliderContent>
+                    <StyledSlider
+                        min={0}
+                        max={100}
+                        value={value}
+                        getAriaValueText={valuetext}
+                        aria-labelledby='discrete-slider-always'
+                        step={1}
+                        data-testid={ROLLOUT_SLIDER_ID}
+                        marks={marks}
+                        onChange={onChange}
+                        valueLabelDisplay='on'
+                        disabled={disabled}
+                    />
+                </SliderContent>
+                <StyledInputBox>
+                    <StyledTextField
+                        size='small'
+                        type='number'
+                        value={inputValue}
+                        onChange={handleInputChange}
+                        onBlur={handleInputBlur}
+                        onKeyDown={handleKeyDown}
```

Comment:
Why does this one need onBlur and onKeyDown? Shouldn't onChange be enough?

Reply:
If we used only onChange on the input field, the slider would update immediately with each keystroke. This would create a poor user experience:
As a user types "75", the slider would first jump to "7" and then to "75"

Reply:
Right. I see the logic there. I don't understand why the onKeyDown handler is there, though. Shouldn't enter submit the form (as is standard with input fields)? Not a big deal, but I don't think we need it 🤷🏼